c/tskit/tables.c

```diff
@@ -6780,40 +6818,75 @@ static int
 cmp_edge_cl(const void *a, const void *b) {
     const tsk_edge_t *ia = (const tsk_edge_t *) a;
     const tsk_edge_t *ib = (const tsk_edge_t *) b;
-    int ret = (ia->child > ib->child) - (ia->child < ib->child);
+    int ret = (ia->parent > ib->parent) - (ia->parent < ib->parent);
     if (ret == 0)  {
-        ret = (ia->left > ib->left) - (ia->left < ib->left);
+        ret = (ia->child > ib->child) - (ia->child < ib->child);
+        if (ret == 0)  {
+            ret = (ia->left > ib->left) - (ia->left < ib->left);
+        }
     }
     return ret;
 }
 
 /* Squash the edges in the specified array in place. The output edges will
  * be sorted by (child_id, left).
  */
+
 int TSK_WARN_UNUSED
-tsk_squash_edges(tsk_edge_t *edges, size_t num_edges, size_t *num_output_edges)
+tsk_squash_edges(tsk_edge_t *edges, size_t num_edges, tsk_size_t *num_output_edges)
```

Comment:
I should have spotted this before, but if we're changing one of these to ``tsk_size_t`` we should change both. Can you change ``num_edges`` to be a ``tsk_size_t`` please?